src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -1322,16 +1443,39 @@ private static unsafe nuint NarrowUtf16ToAscii_Sse2(char* pUtf16Buffer, byte* pA
             }
             else
             {
-                if ((Sse2.MoveMask(Sse2.AddSaturate(utf16VectorFirst.AsUInt16(), asciiMaskForAddSaturate).AsByte()) & NonAsciiDataSeenMask) != 0)
+                if (Sse2.IsSupported)
                 {
-                    return 0;
+                    if ((Sse2.MoveMask(Sse2.AddSaturate(utf16VectorFirst.AsUInt16(), asciiMaskForAddSaturate).AsByte()) & NonAsciiDataSeenMask) != 0)
+                    {
+                        return 0;
+                    }
+                }
+                else
+                {
+                    if (AdvSimd.Arm64.IsSupported)
+                    {
+                        if ((AdvSimd.Arm64.MaxAcross(AdvSimd.AddSaturate(utf16VectorFirst.AsUInt16(), asciiMaskForAddSaturate).AsByte()).ToScalar() & NonAsciiDataSeenMask) != 0)
+                        {
+                            return 0;
+                        }
+                    }
                 }
             }
 
             // Turn the 8 ASCII chars we just read into 8 ASCII bytes, then copy it to the destination.
 
-            Vector128<byte> asciiVector = Sse2.PackUnsignedSaturate(utf16VectorFirst, utf16VectorFirst);
-            Sse2.StoreScalar((ulong*)pAsciiBuffer, asciiVector.AsUInt64()); // ulong* calculated here is UNALIGNED
+            Vector128<byte> asciiVector;
+            if (Sse2.IsSupported)
+            {
+                asciiVector = Sse2.PackUnsignedSaturate(utf16VectorFirst, utf16VectorFirst);
+                Sse2.StoreScalar((ulong*)pAsciiBuffer, asciiVector.AsUInt64()); // ulong* calculated here is UNALIGNED
+            }
+            else
+            {
+                Vector64<byte> tmp = AdvSimd.ExtractNarrowingSaturateUnsignedLower(utf16VectorFirst);
+                asciiVector = AdvSimd.ExtractNarrowingSaturateUnsignedUpper(tmp, utf16VectorFirst).AsByte();
+                AdvSimd.Store((ulong*)pAsciiBuffer, asciiVector.AsUInt64());
```

Comment:
I got a comment in one of my reviews to include the same comment as in `Sse2.StoreScalar`, since it applies here too.
`// ulong* calculated here is UNALIGNED`
This suggestion probably also applies to line 1503, 1687 and 1860.